### PR TITLE
Improve std::thread::Result documentation

### PR DESCRIPTION
**Pull request by golddranks**
Thanks to @dtolnay for pointing out the different premise of the contents of the `Err` variant in `std::thread::Result` WRT normal error handling.

### CODE REVIEW

File: src/libstd/thread/mod.rs [hunk 1]
```diff
@@ -1271,6 +1271,18 @@ impl fmt::Debug for Thread {
 ///
 /// Indicates the manner in which a thread exited.
 ///
+/// The value contained in the `Result::Err` variant
+/// is the value the thread panicked with;
+/// that is, the parameter the `panic!` macro was called with.
```
Comment (Centril) [inline src/libstd/thread/mod.rs]:
```suggestion
/// that is, the argument the `panic!` macro was called with.
```

File: src/libstd/thread/mod.rs [hunk 2]
```diff
@@ -1271,6 +1271,18 @@ impl fmt::Debug for Thread {
...
+/// Unlike with normal errors, this value doesn't implement
+/// the `std::error::Error` trait.
```
Comment (Centril) [inline src/libstd/thread/mod.rs]:
```suggestion
/// the [`Error`] trait.
```
(+ add link to the trait)

File: src/libstd/thread/mod.rs [hunk 3]
```diff
@@ -1271,6 +1271,18 @@ impl fmt::Debug for Thread {
...
+///
+/// Thus, a sensible way to handle a thread panic is to either
+/// `unwrap` the `Result`, propagating the panic,
```
Comment (Centril) [inline src/libstd/thread/mod.rs]:
```suggestion
/// `unwrap` the `Result<T>`, propagating the panic,
```

File: src/libstd/thread/mod.rs [hunk 4]
```diff
@@ -1271,6 +1271,18 @@ impl fmt::Debug for Thread {
...
+/// or in case the thread is intended to be a subsystem boundary
+/// that is supposed to isolate system-level failures,
+/// match for the `Err` variant and handle the panic in an appropriate way.
```
Comment (Centril) [inline src/libstd/thread/mod.rs]:
```suggestion
/// match on the `Err` variant and handle the panic in an appropriate way.
```

File: src/libstd/thread/mod.rs (hunk 3 above)
Comment (Centril) [inline src/libstd/thread/mod.rs]: Style nit... writing in the following way would, I think, be more readable: 

> ...is to either:
> 1. `unwrap`...
> 2. or in the case...

**Comment by Centril** [comment]
r=me rollup with green CI

File: src/libstd/thread/mod.rs (hunk 2 above)
Comment (golddranks) [reply src/libstd/thread/mod.rs]: Ah, sorry, forgot the link!

Comment (Centril) [reply src/libstd/thread/mod.rs]: :)

Comment (golddranks) [reply src/libstd/thread/mod.rs]: Uh, wait, checking how the rest of the stdlib API is doing the linking to `Error`, this *seems* to work as-is.

Comment (Centril) [reply src/libstd/thread/mod.rs]: Curious! The `Error` trait isn't part of the prelude... We'll find out whether CI is happy or not I suppose.

**Comment by golddranks** [comment]
Shoot, it can't resolve it. Fixing.

**Comment by golddranks** [comment]
Hm, I wonder what's wrong here. According to the intra doc links RFC, this should be the correct syntax, and "not being able to resolve" sounds more of a path issue? Maybe `std::error::Error` doesn't work because it's the `std`. Trying with `crate::error::Error`.

**Comment by Centril** [comment]
Seems it worked; thanks! @bors r+ rollup

**Pull request merged**